contributors/design-proposals/resource-class.md

```diff
@@ -0,0 +1,363 @@
+resource reference used in the spec to request the resource and resource reference
+used to advertise the resource availablity.
+
+Since 'CPU' and 'Memory' are resources that are available across all kubernetes
+deployments and need no metadata to describe the resource type any further,
+the current user facing API (Pod Specification) remains portable as long as pod
+requests only CPU and Memory. However the current model cannot support
+complex resources like GPUs, ASICs, NICs, local storage, etc.
+To support heterogeneity, portability and management at scale, such resources
+must be represented(advertised) in a form which allows metadata inclusion and a
+metadata based resource selection mechanism.
+
+_GPU Integration Example:_
+  * [Enable "kick the tires" support for Nvidia GPUs in COS](https://github.com/kubernetes/kubernetes/pull/45136)
+  * [Extend experimental support to multiple Nvidia GPUs](https://github.com/kubernetes/kubernetes/pull/42116)
+
+_Kubernetes Meeting Notes On This:_
+  * [Meeting notes](https://docs.google.com/document/d/1Qg42Nmv-QwL4RxicsU2qtZgFKOzANf8fGayw8p3lX6U/edit#)
+  * [Better Abstraction for Compute Resources in Kubernetes](https://docs.google.com/document/d/1666PPUs4Lz56TqKygcy6mXkNazde-vwA7q4e5H92sUc)
+  * [Extensible support for hardware devices in Kubernetes (join kubernetes-dev@googlegroups.com for access)](https://docs.google.com/document/d/1LHeTPx_fWA1PdZkHuALPzYxR0AYXUiiXdo3S0g2VSlo/edit)
+
+## Use Cases
+
+  * I want to have a compute resource type, 'Resource Class',
+    which can be created with meaningful and portable names. This compute
+    resource can hold additional metadata as well, for example:
+    * `nvidia.gpu.high.mem` is the name and metadata is memory greater than 'X' GB.
+    * `fast.nic` is the name and associated metadata is bandwidth greater than
+      'B' gbps.
+  * If I request a resource `nvidia.gpu.high.mem` for my pod, any 'nvidia-gpu'
+    type device which has memory greater than or equal to 'X' GB, should be able
+    to satisfy this request, independent of other device capabilities such as
+    'version' or 'nvlink locality' etc.
+  * Similarly, if I request a resource `fast.nic`, any nic device with speed
+    greater than 'B' gbps should be able to meet the request.
+  * I want a rich metadata selection interface where operators like 'Eq' for
+    'equals to', 'Lt' for 'less than', 'LtEq' for 'less than equal to', 'Gt' for
+    'greater than', 'GtEq' for 'greater than and equal to' and 'In' for
+    'a set of accepted values' are supported on the compute resource metadata.
+
+## Objectives
+
+1. Define and add support in the API for a new type, *Resource Class*.
+2. Add support for *Resource Class* in the scheduler.
+
+## Non Objectives
+1. Discovery, advertisement, allocation/deallocation of devices is expected to
+   be addressed by [device plugin proposal](https://github.com/kubernetes/community/pull/695/files)
+
+## Resource Class
```

Comment:
I skimmed over the doc. I think any resource class proposal should cover both node resources and cluster resources. This proposal does not cover the latter.